ddtrace/tracer.py

```diff
@@ -14,14 +14,30 @@
 from .sampler import AllSampler, DatadogSampler, RateSampler, RateByServiceSampler
 from .span import Span
 from .utils.formats import get_env
-from .utils.deprecation import deprecated
+from .utils.deprecation import deprecated, warn
 from .vendor.dogstatsd import DogStatsd
 from . import compat
 
 
 log = get_logger(__name__)
 
 
+def _parse_dogstatsd_url(dogstatsd_url):
+    if dogstatsd_url is None:
+        return
+
+    # url can be either of the form `<host>:<port>` or `unix://<path>`
+    if dogstatsd_url.startswith('unix://'):
+        path = dogstatsd_url[len('unix://'):]
+        return dict(socket_path=path)
+    else:
+        try:
+            hostname, port = dogstatsd_url.split(':')
```

Comment:
You say it's an URL and then just treat that as a `hostname:port`?

Anyhow, this is unlikely to work with IPv6.